timeseries_cli_dw.py

```diff
@@ -0,0 +1,203 @@
+}
+TICKER_LISTS_FOLDER = Path("docs/")
+
+
+class DownloadJob(threading.Thread):
+    def __init__(self, tickers: List[str],downloader : TimeseriesDownloader, timeout_time : float, importer : DataImporter):
+        super().__init__()
+        self.shutdown_flag = threading.Event()
+        self.tickers = tickers
+        self.downloader = downloader
+        self.importer = importer
+        self.timeout_time = timeout_time
+
+    def run(self):
+        for ticker in self.tickers:
+            # Actually download data
+            downloaded_data = self.downloader.download_between_dates(
+                ticker=ticker, start=start_date, end=end_date, interval="1m")
+            if(downloaded_data == False):
+                log.e("unable to download {}".format(ticker))
+                time.sleep(self.timeout_time)
+                continue
+            # Upload data
+            log.i("attempting to upload {}".format(ticker))
+            self.importer.from_contents(downloaded_data)
+            log.i("successfully uploaded {}".format(ticker))
+            # Could refactor to wait timeout time - upload time
+            time.sleep(self.timeout_time)
+
+
+def service_shutdown(signum, frame):
+    print('Caught signal %d' % signum)
+    raise ServiceExit
+
+
+class ServiceExit(Exception):
+    """
+    Custom exception which is used to trigger the clean exit
+    of all running threads and the main program.
+    """
+    pass
+
+
+def list_tickers_file(ticker_list_folder: Path) -> Path:
+    '''
+    List json files from the docs folder where to find the ticker list.
+
+    Returns:
+        Path to the selected ticker list file.
+    '''
+    docs_glob = ticker_list_folder.glob('*.json')
+    return [x.name.replace('.json', '') for x in docs_glob if x.is_file()]
+
+
+def retrieve_ticker_list(doc_path: Path) -> List[str]:
+    '''
+    Grabs all tickers from the properly formatted doc_path file.
+
+    Returns:
+        A list of str, names of tickers.
+    '''
+    doc = json.load(doc_path.open("r"))
+    return [ticker['ticker'] for ticker in doc['tickers']]
+
+
+def print_error_msg(msg: str = None):
+    if msg is None:
+        log.e("timeseries_cli_download.py -p <provider: {}> -f <ticker file: {}>".format(
+            list(DOWNLOADERS.keys()), list_tickers_file(TICKER_LISTS_FOLDER)
+        )
+        )
+    else:
+        log.e("{}: timeseries_cli_download.py -p <provider: {}> -f <ticker file: {}>".format(
+            msg,
+            list(DOWNLOADERS.keys()),
+            list_tickers_file(TICKER_LISTS_FOLDER)
+        )
+        )
+
+
+def get_seven_days_ago() -> date:
+    '''
+    Calculates when is 7 days ago.
+
+    Returns:
+        The date of 7 days ago.
+    '''
+    seven_days_delta = timedelta(days=7)
+    seven_days_ago = datetime.now() - seven_days_delta
+    return date(seven_days_ago.year, seven_days_ago.month, seven_days_ago.day)
+
+
+if __name__ == "__main__":
```

Comment:
Codacy found an issue: [If 127 is too complex (16) (MC0001)](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5942194)